middleware/http/opa/middleware.go

```diff
@@ -43,6 +46,26 @@ var (
 	errOpaInvalidResultType = errors.New("got an invalid type back from repo policy. Only a boolean or map is valid")
 )
 
+func (s *Status) UnmarshalJSON(b []byte) error {
+	if len(b) == 0 {
```

Comment:
Given the `omitempty` annotation, I'm okay with rejecting JSON that attempts to define `defaultStatus` without a value ... it simplifies things to just use the default `UnmarshalJSON` to an `interface{}` type to handle edge cases. Does this check handle `"defaultStatus": null` or just `"defaultStatus": []`?

In either case, if you're explicitly handling special cases, I would add the appropriate tests for them in middleware_test.go

Reply:
use json unmarshal to `interface{}` to handle edge cases. Also add some tests.